quasar-core/src/main/java/co/paralleluniverse/concurrent/util/ThreadAccess.java

```diff
@@ -32,50 +37,47 @@
     private static final long inheritableThreadLocalsOffset;
     private static final long contextClassLoaderOffset;
     private static final long inheritedAccessControlContextOffset;
-    private static final Class threadLocalMapClass;
-    private static final Constructor threadLocalMapConstructor;
-    private static final Constructor threadLocalMapInheritedConstructor;
+    private static final Class<?> threadLocalMapClass;
+    private static final Constructor<?> threadLocalMapConstructor;
+    private static final Constructor<?> threadLocalMapInheritedConstructor;
 //    private static final Method threadLocalMapSet;
     private static final Field threadLocalMapTableField;
     private static final Field threadLocalMapSizeField;
     private static final Field threadLocalMapThresholdField;
-    private static final Class threadLocalMapEntryClass;
-    private static final Constructor threadLocalMapEntryConstructor;
+    private static final Class<?> threadLocalMapEntryClass;
+    private static final Constructor<?> threadLocalMapEntryConstructor;
     private static final Field threadLocalMapEntryValueField;
 
     static {
         try {
-            targetOffset = UNSAFE.objectFieldOffset(Thread.class.getDeclaredField("target"));
-            threadLocalsOffset = UNSAFE.objectFieldOffset(Thread.class.getDeclaredField("threadLocals"));
-            inheritableThreadLocalsOffset = UNSAFE.objectFieldOffset(Thread.class.getDeclaredField("inheritableThreadLocals"));
-            contextClassLoaderOffset = UNSAFE.objectFieldOffset(Thread.class.getDeclaredField("contextClassLoader"));
+            targetOffset = UNSAFE.objectFieldOffset(doPrivileged(new GetDeclaredField(Thread.class, "target")));
+            threadLocalsOffset = UNSAFE.objectFieldOffset(doPrivileged(new GetDeclaredField(Thread.class, "threadLocals")));
+            inheritableThreadLocalsOffset = UNSAFE.objectFieldOffset(doPrivileged(new GetDeclaredField(Thread.class, "inheritableThreadLocals")));
+            contextClassLoaderOffset = UNSAFE.objectFieldOffset(doPrivileged(new GetDeclaredField(Thread.class, "contextClassLoader")));
 
             long _inheritedAccessControlContextOffset = -1;
             try {
-                _inheritedAccessControlContextOffset = UNSAFE.objectFieldOffset(Thread.class.getDeclaredField("inheritedAccessControlContext"));
-            } catch (NoSuchFieldException e) {
+                _inheritedAccessControlContextOffset = UNSAFE.objectFieldOffset(doPrivileged(new GetDeclaredField(Thread.class, "inheritedAccessControlContext")));
+            } catch (PrivilegedActionException e) {
+                Throwable t = e.getCause();
+                if (!(t instanceof NoSuchMethodException)) {
```

Comment:
Is that intended that the original code swallows `NoSuchFieldException` and the new code `NoSuchMethodException`? Maybe there is a reason why the privileged access fishes out a method rather than a field, but it looks a bit fishy to me.

Reply:
Ze oops! :man_facepalming: